src/model/operation/markeroperation.js

```diff
@@ -94,19 +94,17 @@ export default class MarkerOperation extends Operation {
 	 * @inheritDoc
 	 */
 	_execute() {
-		if ( this.oldRange === null && this.newRange === null ) {
-			return { name: this.name, type: 'remove' };
-		}
+		// Make a change in Document#markers only when something actually changes, that is
+		// `this.oldRange` and `this.newRange` are different.
+		const changeInMarkersCollection =
+			!( this.oldRange === null && this.newRange === null ) &&
+			!( this.oldRange !== null && this.newRange !== null && this.oldRange.isEqual( this.newRange ) );
 
-		const document = ( this.oldRange || this.newRange ).root.document;
-		let type;
+		const document = changeInMarkersCollection && ( this.oldRange || this.newRange ).root.document;
```

Comment:
Can't you move this like to `changeInMarkersCollection` and simplified the code?